### PR TITLE
Improve message deletion

### PR DESCRIPTION
**Pull request by fischerling**
I had a look at our message deletion.
See: #255

## Result:

* Refactor Message.delete().
* Don't flush the message cache.
* Introduce a trash folder, specified in `global.mail-trash`.

### CODE REVIEW

**Comment by skx** [comment]
Minor comments added, but those aside it looks good at first pass.

File: lumail2.lua [hunk 1]
```diff
@@ -1468,72 +1468,45 @@ end
...
-    -- Get the list of messages, and the current offset
-    -- that'll let us find the message.
-    local offset = Config.get_with_default("index.current", 0)
-    local msgs = get_messages()
-    if not msgs then
-      warning_msg "There are no messages!"
-      return
-    end
+  -- Move message to trash.
+  local trash = Config:get "global.trash-mail"
```
Comment (fischerling) [inline lumail2.lua]: But what should be the default ? Prefix+"Trash" ? What if this Maildir not exists ?

Comment (skx) [inline lumail2.lua]: Default should be unset - i.e. delete does a real delete.

Comment (fischerling) [inline lumail2.lua]: Config:get returns nil if the value is not set.
So what is the difference between get() and get_with_default("", nil) ?

If "global.trash-mail" is unset get returns nil and we do a real delete.

Comment (skx) [inline lumail2.lua]: In practice it doesn't matter, but I think it reads more cleanly.

As it was I applied it, unchanged. :)

Comment (fischerling) [inline lumail2.lua]: fixed in #265.

**Pull request merged**